doc: add release notes for #22729

vasild opened the pull request:
Add release notes for #22729.

vasild committed:
doc: add release notes for #22729

maflcko reviewed (commented):
left some nits/questions

vasild commented on doc/release-notes-22729.md:
```diff
@@ -0,0 +1,17 @@
+P2P and network changes
+-----------------------
+
+- Previously, if Bitcoin Core would be listening for P2P connections, then
```
Done.

vasild commented on doc/release-notes-22729.md:
```diff
@@ -0,0 +1,17 @@
+
+- Previously, if Bitcoin Core would be listening for P2P connections, then
+  it would always bind on `127.0.0.1:8334` to listen for incoming Tor
+  connections. It was not possible to switch off this even if the node didn't
```
Done.

vasild commented on doc/release-notes-22729.md:
```diff
@@ -0,0 +1,17 @@
+- Previously, if Bitcoin Core would be listening for P2P connections, then
+  it would always bind on `127.0.0.1:8334` to listen for incoming Tor
+  connections. It was not possible to switch off this even if the node didn't
+  use Tor. For example, a configuration `bind=addr:port` would result in
```
Applied the `s/For example/Previously/` change. Initially I wrote "for example `1.2.3.4:8333`" but changed it to "for example `addr:port`", now this covers all is not an example anymore.

vasild commented:
`620bbc77d9...307659733e`: address suggestions

vasild commented on doc/release-notes-22729.md:
```diff
@@ -0,0 +1,17 @@
+
+  If you are using `bind=...` without `bind=...=onion` and rely on the previous
+  behavior to accept incoming Tor connections at `127.0.0.1:8334`, you need to
+  make this explicit by using `bind=... bind=127.0.0.1:8334=onion`. (#22729)
```
Done with some minor changes. Thanks!

vasild commented:
`307659733e...a7f5d188cc`: address suggestions

davidgumberg commented:
ACK https://github.com/bitcoin/bitcoin/commit/a7f5d188cc7a23b1946d48693ada22fe90d9ffe0

Nit: Documentation in [`doc/tor.md`](https://github.com/bitcoin/bitcoin/blob/9774a958b501a6d439a734e18b29e04f59f973f6/doc/tor.md?plain=1#L196) should also be updated:

```diff
 (obviously, replace the .onion address with your own). It should be noted that you still
 listen on all devices and another node could establish a clearnet connection, when knowing
 your address. To mitigate this, additionally bind the address of your Tor proxy:

-    ./bitcoind ... -bind=127.0.0.1
+    ./bitcoind ... -bind=127.0.0.1:8334=onion
```

davidgumberg committed:
doc: tor.md: use -bind=127.0.0.1:8334=onion for the Tor bind

After https://github.com/bitcoin/bitcoin/pull/22729 if
`-bind=127.0.0.1` is used, then bitcoind will not automatically
also bind on `127.0.0.1:8334`.

vasild commented:
`a7f5d188cc..6d33e13bd4`: append one commit with that update to `doc/tor.md`, thanks!

davidgumberg commented:
reACK https://github.com/bitcoin/bitcoin/pull/30502/commits/6d33e13bd493ca6cee7b52b990e4822a28e35d0a

willcl-ark reviewed (approved):
ACK 6d33e13bd493ca6cee7b52b990e4822a28e35d0a

Release notes look good to me now, thanks!
